src/native/mono/monodroid/monodroid-glue.cc

```diff
@@ -1551,7 +1558,11 @@ MonodroidRuntime::Java_mono_android_Runtime_initInternal (JNIEnv *env, jclass kl
 
 #if defined (RELEASE)
 	if (application_config.marshal_methods_enabled) {
-		xamarin_app_init (env, get_function_pointer_at_runtime);
+		if (!application_config.managed_marshal_methods_lookup_enabled) {
```

Comment:
"ditto" here as well: there shouldn't be an `else`, so we could just merge things:

```cpp
if (application_config.marshal_methods_enabled &&
        !application_config.managed_marshal_methods_lookup_enabled) {
    xamarin_app_init (env, get_function_pointer_at_runtime);
}
```